rootfs/etc/nginx/lua/sticky.lua

```diff
@@ -0,0 +1,110 @@
+local json = require('cjson')
+local string = require("resty.string")
+local sha1_crypto = require("resty.sha1")
+local md5_crypto = require("resty.md5")
+
+local sticky_sessions = ngx.shared.sticky_sessions
+
+local DEFAULT_SESSION_COOKIE_NAME = "route"
+local DEFAULT_SESSION_COOKIE_HASH = "md5"
+-- Currently STICKY_TIMEOUT never expires
+local STICKY_TIMEOUT = 0
+
+local _M = {}
+
+local sha1 = sha1_crypto:new()
+if not sha1 then
+  ngx.say("failed to create the sha1 object")
+  return
+end
+
+local md5 = md5_crypto:new()
+if not md5 then
+  ngx.say("failed to create the md5 object")
+  return
+end
+
+local function md5_digest(raw)
+  md5:update(raw)
+  return string.to_hex(md5:final())
+end
+
+local function sha1_digest(raw)
+  sha1:update(raw)
+  return string.to_hex(sha1:final())
+end
+
+local function get_cookie_name(backend)
+  local name = backend["sessionAffinityConfig"]["cookieSessionAffinity"]["name"]
+  return name or DEFAULT_SESSION_COOKIE_NAME
+end
+
+local function is_valid_upstream(backend, address, port)
+  for _, ep in ipairs(backend.endpoints) do
+    if ep.address == address and ep.port == port then
+      return true
+    end
+  end
+  ngx.log(ngx.INFO, "session upstream no longer valid, resetting")
+  return false
+end
+
+function _M.is_sticky(backend)
+  return backend["sessionAffinityConfig"]["name"] == "cookie"
+end
+
+function _M.get_upstream(backend)
```

Comment:
I'd like to be more consistent with naming... reading the implementation this function takes a `backend` object and current request and returns an `endpoint` object or `nil`. I suggest we use endpoint instead of upstream. What do you think?